diff --git a/third_party/xla/third_party/tsl/tsl/platform/default/build_config.bzl b/third_party/xla/third_party/tsl/tsl/platform/default/build_config.bzl
--- a/third_party/xla/third_party/tsl/tsl/platform/default/build_config.bzl
+++ b/third_party/xla/third_party/tsl/tsl/platform/default/build_config.bzl
@@ -8,1 +8,0 @@
-load("@com_google_protobuf//:protobuf.bzl", "proto_gen")
@@ -13,0 +12,18 @@
 load("//tsl/platform:build_config_root.bzl", "if_static")
+def proto_gen(name, srcs, outs = [], gen_cc = 0, includes = [], **kwargs):
+    protoc = "@com_google_protobuf//:protoc"
+    cmd = "$(location " + protoc + ") --cpp_out=$(RULEDIR) "
+    cmd += "-I. -I$(RULEDIR) "
+    src_locations = []
+    for s in srcs:
+        if type(s) == "string" and s.startswith(":"):
+            src_locations.append("$(location " + s + ")")
+        else:
+            src_locations.append("$(locations " + s + ")")
+    cmd += " ".join(src_locations)
+    native.genrule(
+        name = name,
+        srcs = srcs,
+        outs = outs,
+        cmd = cmd,
+        tools = [protoc],
+    )
