--- a/tensorflow_metadata/proto/v0/BUILD
+++ b/tensorflow_metadata/proto/v0/BUILD
@@ -13,8 +13,6 @@
 # limitations under the License.
 # ==============================================================================
 
-
-load("@com_google_protobuf//bazel:py_proto_library.bzl", "py_proto_library")
 
 licenses(["notice"])  # Apache 2.0
 
@@ -60,9 +58,12 @@
     deps = [":cc_metadata_v0_proto_cc"],
 )
 
-py_proto_library(
+alias(
     name = "metadata_v0_proto_py_pb2",
-    deps = [
-        ":metadata_v0_proto",
-    ],
+    actual = ":metadata_v0_proto",
 )
+
+alias(
+    name = "metadata_v0_proto_py_pb2_genproto",
+    actual = ":metadata_v0_proto",
+)
